### PR TITLE
feat: Fail fast on unsupported Action parameters. (#161)

### LINKED ISSUE #161: feat: Fail fast on unsupported Action parameters.
- Some Action parameters support only one of a few choises (`true`/`false`, or ...)
- But current codes cause no errors for an unsupported value and will silently continue
- This makes more difficult to find potential issues

### PR DESCRIPTION
**Pull request by MinoruSekine**
Closes #161

## Summary by CodeRabbit

* **Tests**
  * Added parameter validation to ensure correct configuration values are provided.
  * Added edge case testing for unsupported parameter scenarios.

These updates improve the robustness of the action by catching invalid configurations early and testing boundary conditions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/MinoruSekine/setup-scoop/pull/164?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: .github/workflows/_edge_case.yml [hunk 1]
```diff
@@ -29,6 +29,45 @@ jobs:
...
+
+    strategy:
+      matrix:
+        os: [windows-2025, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - if: inputs.target == 'pr'
+        uses: actions/checkout@v6
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply .github/workflows/_edge_case.yml:44]: It loses maintainability to pin version of `actions/checkout` by commit hash. So I accept risk not to pin.

I'll work for `persist-credentials: false` in this comment.

File: .github/workflows/_edge_case.yml [hunk 2]
```diff
@@ -29,6 +29,45 @@ jobs:
...
+        if: inputs.target == 'pr'
+        name: Run actions locally
+        uses: ./
+        with:
+          scoop_checkup: force
+        continue-on-error: true
+
+      - id: dev
+        if: inputs.target == 'dev'
+        uses: MinoruSekine/setup-scoop@main
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply .github/workflows/_edge_case.yml:58]: The purpose of that job is warranty of behavior on dev branch (`main`).
If pin the version by commit hash, we have to update that pinned hash for every merge to `main`. That's not practical, and that update will change commit hash of head of `main` (endless :-) )

File: .github/workflows/_edge_case.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply .github/workflows/_edge_case.yml:44]: @coderabbitai I've resolved about `persist-credential:` in e1d500f.
